### PR TITLE
Add a recommendations doc

### PR DESCRIPTION
**Pull request by jsumners**
This PR adds a document where we can add best practices and recommendations. To start, I have added a recommendation for issues #1996, https://github.com/fastify/help/issues/127, https://github.com/fastify/help/issues/98 and other issues like them.

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: docs/Recommendations.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+# This one serves requests for `https://two.example.com`
+backend example2-backend
+  server example2-1 10.10.12.2:80
+  server example2-2 10.10.12.2:80
+  server example2-3 10.10.12.3:80
+
+# This backend handles the static resources requests.
+backend static-backend
+  server static-server1 10.10.9.2:80
+```
```
Comment (allevo) [inline docs/Recommendations.md]: I move this configuration in another file

Comment (jsumners) [reply docs/Recommendations.md]: That would make the recommendation hard to follow. The comments in this conf are part of the explanation. I think the proper course of action is to have _each_ recommendation in its own markdown file within a recommendations directory. But with only one recommendation at this time, I’d say that should happen with the next one added.

File: docs/Recommendations.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+Python, one would need either a web server with specific support for the
+language or the ability to setup some sort of [CGI gateway][cgi] that works
+with the language. With Node.js, one can simply write an application that
+_directly_ handles HTTP requests. As a result, the temptation is to write
+applications that handle requests for multiple domains, listen on multiple
+ports (i.e. HTTP _and_ HTTPS), and various other scenarios and combinations
+thereof. Further, the temptation is to then expose these applications directly
+to the Internet to handle requests.
+
+The Fastify **strongly** considers this to be an anti-pattern and extremely
```
Comment (mcollina) [inline docs/Recommendations.md]: The Fastify team

**Comment by RafaelGSS** [comment]
I think that we can add a ref [here](https://github.com/fastify/fastify/blob/master/docs/Write-Plugin.md#how-to-write-a-good-plugin)

File: docs/Recommendations.md (hunk 1 above)
Comment (allevo) [reply docs/Recommendations.md]: Yep. But if we would add another configuration? for example nginx

Comment (jsumners) [reply docs/Recommendations.md]: My suggestion would be to add some text like "for an alternative example based on Nginx, see `[nginx.conf][confs/nginx.conf]`"

**Comment by jsumners** [comment]
> I would mention:
>
> nginx
> letsencrypt

I disagree. First, I think certificate management it outside the scope of this example. Second, the Medium article I suggested they read for _why_ they should be doing this in the first place uses Nginx as its example proxy.

I see this doc as it is written to be a straight forward annotated example only. It is heavily implied that the reader should do further research to apply it to their own situation and environment.

**Comment by jsumners** [comment]
@RafaelGSS this document doesn't really have anything to do with plugins or how to write them. Why would a reference be added to the "How to write a good plugin" doc?

File: docs/Recommendations.md (hunk 2 above)
Comment (jsumners) [inline docs/Recommendations.md]:
```suggestion
The Fastify team **strongly** considers this to be an anti-pattern and extremely
```

**Comment by RafaelGSS** [comment]
> @RafaelGSS this document doesn't really have anything to do with plugins or how to write them. Why would a reference be added to the "How to write a good plugin" doc?

In "How to write a good plugin" we suggest most things, maybe reference this section can be a good north for users

**Comment by jsumners** [comment]
@RafaelGSS sorry, I just do not see how the two relate. This example is about how to proxy requests to a fully service. The document you are asking for a reference in is completely geared toward writing a small component that would be used in a full service.

**Comment by jsumners** [comment]
> LGTM as is. We can add more recommendations in the future.

That's definitely my intention with this:

1. Solve for the question we repeatedly get
2. Start _something_ for others to be able to build upon

p.s. I wrote the config that I know. I would have to do research on writing the equivalent Nginx configuration.

File: docs/Recommendations.md (hunk 1 above)
Comment (jsumners) [reply docs/Recommendations.md]: @allevo are you okay merging as-is with these recommendations applied in the future?

**Comment by delvedor** [comment]
Great work @jsumners, this is very useful!

**Comment by jsumners** [comment]
I was waiting for @allevo to respond on our discussion about his concerns, but thank you for merging.
